src/passes/referencedLibraries.ts

```diff
@@ -0,0 +1,34 @@
+import assert = require('assert');
+import { ContractDefinition, ContractKind, FunctionCall, MemberAccess } from 'solc-typed-ast';
+import { AST } from '../ast/ast';
+import { ASTMapper } from '../ast/mapper';
+
+export class ReferencedLibraries extends ASTMapper {
+  visitFunctionCall(node: FunctionCall, ast: AST): void {
+    if (node.vExpression instanceof MemberAccess) {
+      const contractDef = new Map();
+
+      node.vExpression.context?.map.forEach((astNode, key) => {
+        if (astNode instanceof ContractDefinition && astNode.kind === ContractKind.Library) {
+          contractDef.set(key, astNode);
+        }
+      });
+
+      const func_id = node.vExpression.referencedDeclaration;
+
+      contractDef.forEach((astNode, key) => {
+        assert(astNode instanceof ContractDefinition);
+        if (astNode.vFunctions.some((func) => func.id === func_id)) {
+          const parent = node.getClosestParentByType(ContractDefinition);
+
+          if (parent === undefined) return;
```

Comment:
This brings up an interesting point, I was mistaken in believeing libraries could not be called from free functions. Handling that will be a separate PR I think, for now this should throw a NotSupportedYet exception if the library function call is not in a contract